io/src/test/scala/fs2/io/file/BaseFileSpec.scala

```diff
@@ -23,6 +24,9 @@ class BaseFileSpec extends Fs2Spec {
   protected def modify(file: Path): Stream[IO, Unit] =
     Stream.eval(IO(Files.write(file, Array[Byte](0, 1, 2, 3))).void)
 
+  protected def modifyLater(file: Path): Stream[IO, Unit] =
+    Stream.eval(IO(Future(List[Byte](0, 1, 2, 3).foreach(waitAndAppend(file)))).void)
```

Comment:
Even in tests, we should avoid side effects. Instead of writing this to use `Future` and `foreach`, let's write it as a stream of time delayed chunks that writes to a file. Something like this:

```scala
val s = Stream.range(0, 50).map(i => Chunk.array(Array.fill[Byte](1024)(i.toByte))).covary[IO]
s.chunks.metered(25.millis).to(file.writeAll(...)
'''

Reply:
Thanks for the feedback. Tell me if the new version is alright. It changed a bit from the old version, the two streams need to run concurrently...